source/agora/node/BlockStorage.d

```diff
@@ -1055,40 +1055,22 @@ private void testStorage (IBlockStorage storage)
     Transaction[] last_txs;
     Transaction[] txs;
     Block last_block;
-
+    auto signed = BitField!ubyte(6);
+    iota(0, 5).each!(i => signed[i] = true); // last validator does not sign
     void genBlocks (size_t count)
     {
-        if (!last_txs.length)
-        {
-            txs = genesisSpendable().map!(txb => txb.sign()).array();
-            last_block = makeNewTestBlock(blocks[$ - 1], txs);
-            last_txs = txs;
-
-            blocks ~= last_block;
-            block_hashes ~= hashFull(last_block.header);
-            storage.saveBlock(last_block);
-
-            count--;
-        }
-
         while (--count)
         {
-            txs = last_txs.map!(tx => TxBuilder(tx).sign()).array();
-            Enrollment[] no_enrollments = null;
-            uint[] no_missing_validator = null;
-            last_block = makeNewTestBlock(blocks[$ - 1], txs, Hash.init,
-                no_enrollments,
-                no_missing_validator,
-                genesis_validator_keys[0 .. $ - 1], // last validator will not sign
-                (PublicKey k)
-                {
-                    return 0;
-                });
+            txs = last_txs.length ? last_txs.map!(tx => TxBuilder(tx).sign()).array()
+                : genesisSpendable().map!(txb => txb.sign()).array();
+            last_block = makeNewTestBlock(blocks[$ - 1], txs);
+            auto signed_block = last_block.updateSignature(
+                Signature.fromString("0x0000000000000000000000000000000000000000000000000000000000000001" ~
+                    "0000000000000000000000000000000000000000000000000000000000000002"), signed);
```

Comment:
So this works because signatures are not verified, right ?

Reply:
This test is not to verify the signatures but to verify that the block header is updated.